### PR TITLE
refactor: have ESLint manage our JSDocs

### PR DESCRIPTION
**Pull request by michaelfig**
This is much stricter ESLint checking for everything that uses it (except for SwingSet, #1749), but still not as strict as we need: https://github.com/gajus/eslint-plugin-jsdoc/issues/637

### CODE REVIEW

**Review by Chris-Hibbert — APPROVED** [review]
The zoe javascript changes look good. I don't know enough about the .json or yarn.locak changes to opine.

**Review by FUDCo — APPROVED** [review]
Looks fine.  How does this relate to #1749?

**Review by erights — APPROVED** [review]
LGTM except for the removal of spaces around the double ampersand. Why? To me it makes the code substantially harder to read.

File: packages/cosmic-swingset/lib/ag-solo/vats/bridge.js [hunk 1]
```diff
@@ -1,11 +1,10 @@
-/* global harden */
 // @ts-check
 
 import makeStore from '@agoric/store';
 
 /**
  * @template T
- * @typedef {Object} Device
+ * @typedef {'Device' & { __deviceType__: T }} Device
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/cosmic-swingset/lib/ag-solo/vats/bridge.js:7]: Not strictly necessary, just to remind people not to try actually using `d.__deviceType__`.  Do you have a better suggestion?

File: packages/dapp-svelte-wallet/api/package.json [hunk 2]
```diff
@@ -6,10 +6,11 @@
   "scripts": {
     "build": "exit 0",
     "test": "ava",
-    "lint": "yarn lint:types && eslint '**/*.js'",
+    "lint": "yarn lint:types&&yarn lint:eslint",
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/dapp-svelte-wallet/api/package.json]: Windows experience.  Okay, I've reverted that.

**Comment by michaelfig** [comment]
> Looks fine. How does this relate to #1749?

Getting SwingSet to pass the linter (#1749) was such a huge change that I expected it would take longer to land than this broader one.  So, I decoupled it.

**Review by katelynsills — COMMENTED** [review]
LGTM! sorry for the late review

**Pull request merged**